src/prometheus/prometheus.service.ts

```diff
@@ -0,0 +1,40 @@
+import { Cron, CronExpression } from '@nestjs/schedule';
+import { PrismaService } from 'src/db/prisma.service';
+import { AggregateMetrics } from './aggregate-metrics';
+import { AllocatorReportGeneratorMetrics } from './allocator-metrics';
+import { ClientReportGeneratorMetrics } from './client-metrics';
+import { PgPoolMetrics } from './db-metrics';
+import { Injectable } from '@nestjs/common';
+
+@Injectable()
+export class PrometheusMetricService {
+  constructor(
+    private readonly prisma: PrismaService,
+    public readonly clientReportGeneratorMetrics: ClientReportGeneratorMetrics,
+    public readonly allocatorReportGeneratorMetrics: AllocatorReportGeneratorMetrics,
+    public readonly pgPoolMetrics: PgPoolMetrics,
+    public readonly aggregateMetrics: AggregateMetrics,
+  ) {}
+
+  @Cron(CronExpression.EVERY_10_SECONDS)
+  async updateMetrics() {
+    try {
+      const result = await this.prisma.$queryRaw<
+        { active: number; total: number }[]
+      >`
+        SELECT 
+          COUNT(*) FILTER (WHERE state = 'active') AS active,
+          COUNT(*) AS total
+        FROM pg_stat_activity;
+      `;
```

Comment:
Nah, this will show all connections opened on the database - even those opened by other apps or aws itself etc. This is already tracked by cloudwatch, we don't need to track it on the application side.

What we need is data from the application clients:
* `prisma_pool_connections_*` metrics from https://www.prisma.io/docs/orm/prisma-client/observability-and-logging/metrics
* properties of `postgresService.pool` - `totalCount`, `idleCount` - https://node-postgres.com/apis/pool#properties